### DISCUSSION TITLE
Suggestion: Remove mod objects that does not exist anymore

### DISCUSSION THREAD

**Discussion by StolleJay** [voxeltycoon/issues#454] [Modding]
It would be really helpfull if a mod object or something does not exists anymore but its already placed in the savegame that this mod object will automatically removed or replaced with a "dummy" so you can still load and play your savegame.

**Comment by victorleblais** [comment]
Bouncing on that: if you subscribe to a mod that adds assets, and you play with that, and later on the mod updates and removes or renames some assets, it will make your game incompatible.

I don't know what's the best way to handle that. Replacing the 3D model with a dummy is one thing, but you don't have the info about the object's properties (for example, if it's a truck: the speed, the capacities, the cargo it can carry).

Of the top of my head, I can only think of caching the used assets so that if a mod is update or is removed, you use the cached asset, but for example you can make them bright red and/or warn the user to get rid of those buildings/whatever. Don't know how it would apply to mods that change more than just assets.

There's probably a lot of issues that I'm not covering.
